### PR TITLE
Onchain Tests for BeaconChainProofs.sol

### PR DESCRIPTION
**Pull request by Sidu28**
Added 2 tests for ProveValidatorContainers and ProveWithdrawals that test it against this goerli deployment of the beaconchainproofs.sol:

https://goerli.etherscan.io/address/0xcb5e2cbd8df189aff1e94cf471a869e220e95c85#code

### CODE REVIEW

File: build/BeaconChainProofs.abi [hunk 1]
```diff
@@ -0,0 +1,138 @@
+[
```
Comment (gpsanant) [inline build/BeaconChainProofs.abi:1]: remove

File: onchain_tests/onchain_test.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
+package onchain_tests
```
Comment (gpsanant) [inline onchain_tests/onchain_test.go:1]: no need for separate folder?

File: onchain_tests/onchain_test.go [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+	log.Println("Tearing down suite")
+	teardownSuite()
+
+	// Exit with test result code
+	os.Exit(code)
+}
+
+func setupSuite() {
+	RPC := "https://rpc.ankr.com/eth_goerli"
+	PrivateKey := os.Getenv("PRIVATE_KEY")
```
Comment (gpsanant) [inline onchain_tests/onchain_test.go:62]: make all lowercase/camelcase

File: onchain_tests/onchain_test.go [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+func TestWithdrawalProofOnChain(t *testing.T) {
+
+	versionedOracleState, err := beacon.CreateVersionedState(&oracleState)
+	if err != nil {
+		fmt.Println("error creating versioned state", err)
+		return
+	}
+
+	historicalSummaryStateJSON, err := eigenpodproofs.ParseJSONFileDeneb("../data/deneb_goerli_slot_7421952.json")
```
Comment (gpsanant) [inline onchain_tests/onchain_test.go:181]: could you do a capella one too?

**Review by gpsanant — APPROVED** [review]
looks great, can you also just add capella?

File: onchain_tests/onchain_test.go (hunk 4 above)
Comment (Sidu28) [reply onchain_tests/onchain_test.go:182]: added

**Pull request merged**